packages/eslint-plugin/tests/rules/no-deprecated.test.ts

```diff
@@ -691,6 +898,43 @@ exists('/foo');
         },
       ],
     },
+    {
+      code: `
+        const a = {
+          /** @deprecated */ b: 1,
+        };
+        const key = 'b';
+        a[key];
+      `,
+      errors: [
+        {
+          column: 11,
+          data: { name: 'key' },
```

Comment:
[Bug] `key` isn't deprecated, `b` is. TypeScript correctly reports `'b' is deprecated.` so the rule should too. 

https://deploy-preview-11006--typescript-eslint.netlify.app/play/#ts=5.8.2&fileType=.ts&code=MYewdgzgLgBAhjAvDA3gKBjA9AKhzAAQBMBTABwCcTg4oSiYcsMYAjALhgEYAaNAXwDcaNKEiwA1iQCeSGAHJW84WjgBtKdIC6goA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

(here and elsewhere)